integrations/transportation/ryanair.js

```diff
@@ -0,0 +1,141 @@
+import moment from 'moment';
+import request from 'superagent';
+import { ACTIVITY_TYPE_TRANSPORTATION, TRANSPORTATION_MODE_PLANE } from '../../definitions';
+import { HTTPError } from '../utils/errors';
+
+const agent = request.agent();
+const BASE_URL = 'https://api.ryanair.com/userprofile/rest/api/v1/';
+const LOGIN_URL = `${BASE_URL}login`;
+const PROFILE_URL = `${BASE_URL}secure/users/`;
+// user info available at `${PROFILE_URL}${customerId}/profile/full/`
+
+async function logIn(username, password) {
+  const res = await agent
+    .post(LOGIN_URL)
+    .type('application/x-www-form-urlencoded')
+    .send({
+      password,
+      username,
+    });
+
+  if (!res.ok) {
+    throw HTTPError('Error while logging in.');
+  }
+
+  return {
+    token: res.body.token,
+    customerId: res.body.customerId,
+  };
+}
+
+async function connect(requestLogin, requestWebView) {
+  // Here we can request credentials etc..
+
+  // Here we can use two functions to invoke screens
+  // requestLogin() or requestWebView()
+  const { username, password } = await requestLogin();
+
+  if (!(password || '').length) {
+    throw HTTPError('Password cannot be empty');
+  }
+
+  return logIn(username, password);
+}
+
+function disconnect() {
+  // Here we should do any cleanup (deleting tokens etc..)
+  return {};
+}
+
+async function getAllFlights(bookings, customerId, token) {
+  const allFlights = [];
+  try {
+    await Promise.all(bookings.map(async (entry) => {
+      await agent
+        .put(`${PROFILE_URL}${customerId}/bookings/booking`)
+        .set('Accept', 'application/json')
+        .set('X-Auth-Token', token)
+        .send({
+          bookingId: entry.bookingId,
+          pnr: entry.pnr,
+        })
+        .then((res) => {
+          if (res.ok) {
+            res.body.Flights.forEach((singleFlight) => {
+              allFlights.push({
+                bookingId: res.body.BookingId,
+                flightInfo: singleFlight,
+                pnr: res.body.RecordLocator,
+              });
+            });
+          }
+        });
+    }));
+  } catch (error) {
+    throw HTTPError(`Error while fetching past flights. ${error}`);
```

Comment:
Instead of try/catch you should use if (!res.ok) ... like in `logIn` in order to catch the statuscode, as not all errors will result in an exception being raised.